src/biz/tugay/groovyship/modal/Coordinate.groovy

```diff
@@ -0,0 +1,34 @@
+package biz.tugay.groovyship.modal
+
+import groovy.transform.Memoized
+
+/**
+ * A Coordinate represents a certain point.
+ *
+ * <pre>
+ * 0,0  5,0
+ * .....
+ * .
+ * .
+ * .
+ * .
+ * .
+ * 0,5
+ * </pre>
+ */
+class Coordinate
+{
+  final int column
+
+  final int row
+
+  private Coordinate(int column, int row) {
+    this.column = column
+    this.row = row
+  }
+
+  @Memoized
```

Comment:
This is a beautiful feature of Groovy.

Reply:
what does it do? In python one put @decorator above def function or def class and the decorator will be called first with the annotated method being passed. That way I can manipulate the method signature a bit. The decorator has usually a different method signature than the decorated method (= function)

Reply:
Basically caches the instance before returning, mapping it into the input parameters provided. Subsequent calls with same parameters will not invoke the function and will returned the cached instance immediately. 

This way I do not need to implement `equals` and `hashCode` for `Coordinate` class where I would have compared the `column` and `row` values. I can rely on instances being the same when I compare them, because there will always be a single instance of `Coordinate` object within the JVM for a given `column` and row`.